com.unity.sg2/Editor/GraphDeltaRegistry/Generation/Interpreter.cs

```diff
@@ -201,15 +216,15 @@ internal static void EvaluateGraphAndPopulateDescriptors(NodeHandler rootNode, G
 
             var shaderFunctions = new ShaderFunctionRegistry();
             var includes = new List<ShaderFoundry.IncludeDescriptor>();
-            foreach(var node in GatherTreeLeafFirst(rootNode))
+            foreach(var node in topoList)
```

Comment:
GatherTreeLeafFirst works for a tree, but I don't think it does DAGs in topological order.
https://unity.enterprise.slack.com/files/W0192PQLCEQ/F03RTM449CP/image.png